trino/trino.go

```diff
@@ -1136,6 +1225,104 @@ func (st *driverStmt) exec(ctx context.Context, args []driver.NamedValue) (*stmt
 	return &sr, handleResponseError(resp.StatusCode, sr.Error)
 }
 
+type SegmentFetcher struct {
+	ctx             context.Context
+	httpClient      http.Client
+	spooledMetadata spooledMetadata
+}
+
+func (sf *SegmentFetcher) roundTrip(req *http.Request) (*http.Response, error) {
+	delay := 100 * time.Millisecond
+	const maxDelayBetweenRequests = float64(15 * time.Second)
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+	for {
```

Comment:
There should be an upper bound for the number of retries.

Reply:
I don't remember exactly, but probably because there's a timeout in the context. The difference is that iterating over nextUri cannot be parallelized, so a single timeout is ok. We need a shorter timeout here and handle it's expiration.